tests/ui-api-layer-acceptance-tests/internal/domain/k8s/replicaset_test.go

```diff
@@ -0,0 +1,239 @@
+	require.NoError(t, err)
+
+	t.Log("Retrieving replicaSet...")
+	err = waiter.WaitAtMost(func() (bool, error) {
+		_, err := appsV1Client.ReplicaSets(replicaSetNamespace).Get(replicaSetName, metav1.GetOptions{})
+		if err == nil {
+			return true, nil
+		}
+		return false, err
+	}, time.Minute)
+	require.NoError(t, err)
+
+	t.Log("Querying for replicaSet...")
+	var replicaSetRes replicaSetQueryResponse
+	err = c.Do(fixReplicaSetQuery(), &replicaSetRes)
+	require.NoError(t, err)
+	assert.Equal(t, replicaSetName, replicaSetRes.ReplicaSet.Name)
+	assert.Equal(t, replicaSetNamespace, replicaSetRes.ReplicaSet.Namespace)
+
+	t.Log("Querying for replicaSets...")
+	var replicaSetsRes replicaSetsQueryResponse
+	err = c.Do(fixReplicaSetsQuery(), &replicaSetsRes)
+	require.NoError(t, err)
+	assert.Equal(t, replicaSetName, replicaSetsRes.ReplicaSets[0].Name)
+	assert.Equal(t, replicaSetNamespace, replicaSetsRes.ReplicaSets[0].Namespace)
+
+	t.Log("Updating...")
+	replicaSetRes.ReplicaSet.JSON["metadata"].(map[string]interface{})["labels"] = map[string]string{"foo1": "bar1"}
+	update := stringifyJSON(replicaSetRes.ReplicaSet.JSON)
+	var updateRes updateReplicaSetMutationResponse
+	err = c.Do(fixUpdateReplicaSetMutation(update), &updateRes)
+	require.NoError(t, err)
+	assert.Equal(t, replicaSetName, updateRes.UpdateReplicaSet.Name)
+	assert.Equal(t, replicaSetNamespace, updateRes.UpdateReplicaSet.Namespace)
+
+	t.Log("Deleting replicaSet...")
+	var deleteRes deleteReplicaSetMutationResponse
+	err = c.Do(fixDeleteReplicaSetMutation(), &deleteRes)
+	require.NoError(t, err)
+	assert.Equal(t, replicaSetName, deleteRes.DeleteReplicaSet.Name)
+	assert.Equal(t, replicaSetNamespace, deleteRes.DeleteReplicaSet.Namespace)
+
+	t.Log("Waiting for deletion...")
+	err = waiter.WaitAtMost(func() (bool, error) {
+		_, err := appsV1Client.ReplicaSets(replicaSetNamespace).Get(replicaSetName, metav1.GetOptions{})
+		if errors.IsNotFound(err) {
+			return true, nil
+		}
+		if err != nil {
+			return false, err
+		}
+		return false, nil
+	}, time.Minute)
+	require.NoError(t, err)
+
+}
+
+func fixReplicaSet(name, namespace string) *appsv1.ReplicaSet {
+	return &appsv1.ReplicaSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+		Spec: appsv1.ReplicaSetSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"foo": "bar",
+				},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"foo": "bar",
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "foocontainer",
+							Image: "invalid",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func fixReplicaSetQuery() *graphql.Request {
+	query := `query ($name: String!, $namespace: String!) {
+				replicaSet(name: $name, namespace: $namespace) {
+					name
```

Comment:
Please define a new function, which returns ReplicaSet details:

```
					name
					namespace
					creationTimestamp
					labels
					pods
					images
					json
```

We should define them once.